cmake/UserOverride.cmake

```diff
@@ -32,4 +32,9 @@ elseif (CMAKE_CXX_COMPILER_ID MATCHES Clang)
     set(common "-std=c++11 -Wall -Wextra -fPIC")
     set(CMAKE_CXX_FLAGS_RELEASE_INIT "${common} -O3 -march=native -ffast-math -funroll-loops")
     set(CMAKE_CXX_FLAGS_DEBUG_INIT   "${common} -g")
+elseif (CMAKE_CXX_COMPILER_ID STREQUAL "PGI")
+    # pgcpp
+    set(common "--gnu --c++11 -D__GXX_EXPERIMENTAL_CXX0X__")
+    set(CMAKE_CXX_FLAGS_RELEASE_INIT "${common}")
+    set(CMAKE_CXX_FLAGS_DEBUG_INIT   "${common}")
 endif ()
```

Comment:
I tried PGI, it actually passes cmake with this and compiles Teuchos, but fails in SymEngine, it couldn't find `<atomic>`. I think the C++ standard library it had access to was very old, and I didn't figure out how to make it use some newer one. But I figured I'll just leave this in here in case somebody else wants to port it to PGI in the future.